tests/test_csrToCscOffsetMap.cpp

```diff
@@ -0,0 +1,63 @@
+#include <config.h>
+#include <iostream>
+
+#define BOOST_TEST_MODULE CsrToCscOffsetMap
+#include <boost/test/unit_test.hpp>
+#include <boost/version.hpp>
+
+#include <dune/common/version.hh>
+#include <dune/common/fvector.hh>
+#include <dune/istl/bcrsmatrix.hh>
+#include <dune/istl/matrixmarket.hh>
+
+#include <opm/simulators/linalg/bda/BISAI.hpp>
+
+BOOST_AUTO_TEST_CASE(testcsrtocscoffsetmap){
+    using Matrix = Dune::BCRSMatrix<double>;
+
+    Matrix matrix;
+    {
+        std::ifstream mfile("offset_map_matrix.txt");
+        if (!mfile) {
+            throw std::runtime_error("Could not read matrix file");
+        }
+        readMatrixMarket(matrix, mfile);
+    }
+
+    // a transposed version of the matrix is read because the transposed
+    // of a CSR representation is equivalente to CSC
+    Matrix matrix_transposed;
+    {
+        std::ifstream mfile("offset_map_matrix_transposed.txt");
+        if (!mfile) {
+            throw std::runtime_error("Could not read matrix file");
+        }
+        readMatrixMarket(matrix_transposed, mfile);
+    }
+
+    // has to make copy because the output of readMatrixMarket does not
+    // have contiguous non-zero values
+    Matrix matrix_copy(matrix);
+    Matrix matrix_transposed_copy(matrix);
```

Comment:
that means that `matrix_copy` and matrix_transposed_copy` are indentical.

Reply:
Fixed